src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -100,6 +102,74 @@ void SHAMap::visitNodes(std::function<bool (SHAMapAbstractNode&)> const& functio
     }
 }
 
+void
+SHAMap::visitDifferences(SHAMap const* have,
+    std::function<bool (SHAMapAbstractNode&)> function) const
+{
+    // Visit every node in this SHAMap that is not present
+    // in the specified SHAMap
+    assert (root_->isValid ());
+
+    if (! root_)
+        return;
```

Comment:
Over time I believe that we have changed to never allow `root_ == nullptr`.  That being said, I have no problem with this check.  But shouldn't this come prior to the `assert (root_->isValid ());`?  If `root_ == nullptr`, and asserts are enabled, we're going to crash in the assert before we can return.